src/main/scala/com/databricks/labs/overwatch/validation/DeploymentValidation.scala

```diff
@@ -0,0 +1,582 @@
+          s"""Folder should be present from :${fromDT} to:${untilDT} """,
+          Some("SUCCESS"),
+          Some(workspace_id)
+        )
+      } else {
+        val msg =
+          s"""ReValidate the folder existence
+             | Make sure folder with required date folder exist inside ${auditlogprefix_source_aws}/workspaceId=${workspace_id}
+             |, primordial_date:${primordial_date}
+             |, maxDate:${maxDate} """.stripMargin
+        logger.log(Level.ERROR, msg)
+        DeploymentValidationReport(false,
+          getSimpleMsg("Validate_AuditLogPrefix"),
+          s"""Folder should be present from :${fromDT} to:${untilDT} """,
+          Some(msg),
+          Some(workspace_id)
+        )
+      }
+
+    } catch {
+      case exception: Exception =>
+        val msg =
+          s"""AuditLogPrefixTest workspace_id:${workspace_id}
+             | Make sure folder with required date folder exist inside ${auditlogprefix_source_aws}
+             |, primordial_date:${primordial_date}
+             |, maxDate:${maxDate} """.stripMargin
+        logger.log(Level.ERROR, msg)
+        val fullMsg = PipelineFunctions.appendStackStrace(exception, msg)
+        DeploymentValidationReport(false,
+          getSimpleMsg("Validate_AuditLogPrefix"),
+          "Validating folders",
+          Some(fullMsg),
+          Some(workspace_id)
+        )
+    }
+  }
+
+  /**
+   * Performs validation of event hub data for Azure
+   *
+   * @param workspace_id
+   * @param scope
+   * @param key
+   * @param ehName
+   */
+  private def validateEventHub(workspace_id: String, scope: String, key: String, ehName: String, outputPath: String): DeploymentValidationReport = {
+    val testDetails = s"""Connectivity test with ehName:${ehName} scope:${scope} SecretKey_DBPAT:${key}"""
+    try {
+      import org.apache.spark.eventhubs.{ConnectionStringBuilder, EventHubsConf, EventPosition}
+      val ehConn = dbutils.secrets.get(scope = scope, key)
+      //TODO limit 10 check aws
+      //pull 10 milit from aws audit log and grab org id
+      val connectionString = ConnectionStringBuilder(
+        PipelineFunctions.parseAndValidateEHConnectionString(ehConn, false))
+        .setEventHubName(ehName)
+        .build
+
+      val ehConf = EventHubsConf(connectionString)
+        .setMaxEventsPerTrigger(100)
+        .setStartingPosition(EventPosition.fromStartOfStream)
+
+      val rawEHDF = spark.readStream
+        .format("eventhubs")
+        .options(ehConf.toMap)
+        .load()
+        .withColumn("deserializedBody", 'body.cast("string"))
+      val timestamp = LocalDateTime.now(Pipeline.systemZoneId).toInstant(Pipeline.systemZoneOffset).toEpochMilli
+      val tmpPersistDFPath = s"""$outputPath/${ehName.replaceAll("-", "")}/$timestamp/ehTest/rawDataDF"""
+      val tempPersistChk = s"""$outputPath/${ehName.replaceAll("-", "")}/$timestamp/ehTest/rawChkPoint"""
+
+
+      rawEHDF
+        .writeStream
+        .trigger(Trigger.Once)
```

Comment:
Loving this! This will allow us to get a good sample without running forever to get through all the data. Nice work.

Reply:
thanks :)